Task10_Auth/src/test/java/CreateUsersTests.java

```diff
@@ -0,0 +1,108 @@
+        ResponseWrapper getUsersResponse = usersClient.getUsers();
+        ArrayList<UsersDto> usersList = getUsersResponse.getResponseBodyUsers();
+        if (usersList.contains(userToAdd)) {
+            addedUser = usersList.get(usersList.indexOf(userToAdd));
+        }
+        UsersDto finalUserToAdd = userToAdd;
+        UsersDto finalAddedUser = addedUser;
+        assertAll("Status code is 201 // user is added, zip code is removed",
+                () -> assertEquals(exactZipCodesCountBeforeCreation - 1, exactZipCodesCountAfterCreation),
+                //() -> assertTrue(usersList.contains(finalUserToAdd)),
+                () -> assertEquals(finalUserToAdd, finalAddedUser),
+                () -> assertEquals(201, createUserResponse.getResponseCode()));
+    }
+
+    @Test
+    public void createUserWithRequiredFieldsTest() {
+        String name = RandomStringUtils.random(5, true, false);
+        UsersDto.sex userSex = randomSex();
+        UsersDto userToAdd = new UsersDto();
+        userToAdd = userToAdd.createUserDtoObject(name, userSex);
+
+        ResponseWrapper createUserResponse = usersClient.createUser(userToAdd);
+
+        ResponseWrapper getUsersResponse = usersClient.getUsers();
+        ArrayList<UsersDto> usersList = getUsersResponse.getResponseBodyUsers();
+        UsersDto addedUser = new UsersDto();
+        if (usersList.contains(userToAdd)) {
+            addedUser = usersList.get(usersList.indexOf(userToAdd));
+        }
+        UsersDto finalAddedUser = addedUser;
+        UsersDto finalUserToAdd = userToAdd;
+        assertAll("Status code is 201 // user is added",
+                () -> assertEquals(finalUserToAdd, finalAddedUser),
+                () -> assertEquals(201, createUserResponse.getResponseCode()));
+    }
+
+    @Test
+    public void createUserWithIncorrectZipCodeTest() {
+        ResponseWrapper getZipCodesResponse = zipCodesClient.getZipCodes();
+        int age = RandomUtils.nextInt(1, 99);
+        String name = RandomStringUtils.random(5, true, false);
+        UsersDto.sex userSex = randomSex();
+        String zipCode;
+        do {
+            zipCode = RandomStringUtils.random(5, false, true);
+        } while (getZipCodesResponse.getResponseBodyZipCodes().contains(zipCode));
+        UsersDto userToAdd = new UsersDto();
+        userToAdd = userToAdd.createUserDtoObject(age, name, userSex, zipCode);
+
+        ResponseWrapper createUserResponse = usersClient.createUser(userToAdd);
+
+        ResponseWrapper getUsersResponse = usersClient.getUsers();
+        ArrayList<UsersDto> usersList = getUsersResponse.getResponseBodyUsers();
+        UsersDto finalUserToAdd = userToAdd;
+        assertAll("Status code is 424 // user is NOT added",
+                () -> assertFalse(usersList.contains(finalUserToAdd)),
+                () -> assertEquals(424, createUserResponse.getResponseCode()));
+    }
+
+    @Test
+    public void createUserWithDuplicateNameAndSexTest() {
```

Comment:
This test will fail if there are no users in app. Also, when you are trying to add user with unavailable zip code - you will get error.

Reply:
So, I have to check if there are some users created, and if there are no users I should fail this test with some message?

Reply:
You must be independent from existing users and just create them as precondition for your test. When you're testing that you can delete email in some email service - you're creating email before deletion

Reply:
done